src/intervals/arithmetic/basic.jl

```diff
@@ -0,0 +1,302 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+#=  This file contains the functions described as "Basic arithmetic functions"
+    in the IEEE Std 1788-2015 (sections 9.1) and required for set-based flavor
+    in section 10.5.3, at the exception of the `sqr` function present in the
+    "Power functions" file.
+=#
+
++(a::Interval) = a  # Not in the IEEE standard
+
+"""
+    -(a::Interval)
+
+Implement the `neg` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+-(a::F) where {F<:Interval} = F(-a.hi, -a.lo)
+
+
+"""
+    +(a::Interval, b::Real)
+    +(a::Real, a::Interval)
+    +(a::Interval, b::Interval)
+
+Implement the `add` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function +(a::F, b::T) where {T, F<:Interval{T}}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, a.lo + b, a.hi + b)
+end
++(a::Interval{T}, b::S) where {T, S<:Real} = a + Interval{T}(b)
++(b::Real, a::Interval) = a + b
+
+function +(a::F, b::F) where {F<:Interval}
+    (isempty(a) || isempty(b)) && return emptyinterval(F)
+    return @round(F, a.lo + b.lo, a.hi + b.hi)
+end
+
+"""
+    -(a::Interval, b::Real)
+    -(a::Real, a::Interval)
+    -(a::Interval, b::Interval)
+
+Implement the `sub` function of the IEEE Std 1788-2015 (Table 9.1).
+"""
+function -(a::F, b::Real) where {F<:Interval}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, a.lo - b, a.hi - b)
+end
+
+function -(b::Real, a::F) where {F<:Interval}
+    isempty(a) && return emptyinterval(F)
+    return @round(F, b - a.hi, b - a.lo)
+end
+
+function -(a::F, b::F) where {T<:Real, F<:Interval{T}}
+    (isempty(a) || isempty(b)) && return emptyinterval(F)
+    return @round(F, a.lo - b.hi, a.hi - b.lo)
+end
+
+-(a::Interval{T}, b::S) where {T<:Real, S<:Real} = a - Interval{T}(b)
+-(a::T, b::Interval{S}) where {T<:Real, S<:Real} = Interval{S}(a) - b
+
+"""
+    scale(α, a::Interval)
+
+Multiply an interval by a positive scalar.
+
+For efficiency, does not check that the constant is positive.
+"""
+scale(α, a::F) where {F<:Interval}= @round(F, α*a.lo, α*a.hi)
```

Comment:
Few comments/suggestions on this function:
- Is this function used? I don't think so, so we may consider deleting it. (It could be used in the method for `*` in line 81 below.)
- Perhaps we could/should inline it for efficiency
- I think we should `@assert` that α is positive
- I think this function is somewhat flavor dependent (as some edge cases for `*` as remarked below), in the sense of what should it be expected to be the result, e.g. `scale(Inf, some_interval)` or `scale(NaN, some_interval)`. Perhaps this indicates certain flavor dependency of the behavior of `@round` as well.
- Somehow, rounding is not (properly) working here, or is it? `IntervalArithmetic.scale(0.1, Interval(1.0))` returns `Interval(0.1, 0.1)`, which does not contain (the real) 0.1.

Reply:
Answering myself the first question, yes, this function is used in `src/intervals/arithmetic/trigonometric.jl`; so forget that first point.

Reply:
- I think it initially had another name in the trigonometric file.
- I am not sure about `@assert`, the function is not exported so I think it is fine if it is a bit unsafe for efficiency's sake. Not a big deal either way.
- Yes I fear there is a lot of flavor dependent subtleties left and right. I think it is fine to wait for now, and only try to find them all when implementing a second flavor.
- I believe the rounding is correct. In you example `0.1` is already converted to its closest floating point representation before being passed to `scale`. There is nothing we can do about it. It is the same with e.g. `+`, `(0..0) + 0.1` is also `Interval(0.1, 0.1)`.